packages/runtime/src/templates/edge-shared/utils.ts

```diff
@@ -31,8 +31,13 @@ export const addMiddlewareHeaders = async (
   // We need to await the response to get the origin headers, then we can add the ones from middleware.
   const res = await originResponse
   const response = new Response(res.body, res)
+  const originCookies = response.headers.get('set-cookie')
   middlewareResponse.headers.forEach((value, key) => {
     response.headers.set(key, value)
+    // Append origin cookies after middleware cookies
+    if (key == 'set-cookie' && originCookies) {
```

Comment:
```suggestion
    if (key === 'set-cookie' && originCookies) {
```
We should (almost) always use strict comparison